File: syJoh/BOJ/두 수의 합.py
import sys
n=int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))
x = int(sys.stdin.readline())
arr.sort()
answer=0
left, right = 0, n-1

while left<right:
    tmp = arr[left] + arr[right]
    if tmp==x:
        answer+=1
        left+=1
    elif tmp<x:
        left+=1
    else:
        right-=1
print(answer)

"""
# 시간초과
import sys
n=int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))
x = int(sys.stdin.readline())
answer=0

for i in range(n-1):
    for j in range(i+1, n):
        if arr[i]+arr[j]==x:
            answer+=1
            break
print(answer)
"""

"""
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] arr = new int[n];
        for(int i=0; i<n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
        }
        int x = Integer.parseInt(br.readLine());
        int answer = 0;
        for(int i=0; i<n-1; i++){
            for(int j=i+1; j<n; j++){
                if(arr[i]+arr[j]==x){
                    answer++;
                    break;
                }
            }
        }
        br.close();
        System.out.println(answer);
    }
}
"""

File: raehyeon/BOJ/Silver/3/두_수의_합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for(int i=0; i<arr.length; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        int x = Integer.parseInt(br.readLine());
        int count = 0;

        Arrays.sort(arr);
        for(int i=0; i<arr.length; i++) {
            if(arr[i] <= x) {
                for(int j=i+1; j<arr.length; j++) {
                    int num = x - arr[i];
                    if(arr[j] == num) {
                        count++;
                        break;
                    }
                }
            } else {
                break;
            }
        }

        System.out.println(count);
        br.close();
    }
}
